src/main/webapp/dashboard/controller/Master.controller.js

```diff
@@ -39,19 +39,26 @@ sap.ui.define(
           this.oRouter.navTo("detail", oParams);
         },
         onSearch: function (oEvent) {
-          let oTableSearchState = [],
-            sQuery = oEvent.getParameter("query");
+          const sQuery = oEvent.getParameter("query");
+          const aFilters = [];
 
-          if (sQuery && sQuery.length > 0) {
-            oTableSearchState = [
-              new Filter("UserId", FilterOperator.Contains, sQuery),
-            ];
+          // Validate sQuery
+          if (!sQuery || sQuery.length === 0) {
+            return;
           }
-
-          this.getView()
-            .byId("runtimeConfigTable")
-            .getBinding("items")
-            .filter(oTableSearchState, "Application");
+          // Create filters and push only if the query exists
+          aFilters.push(new Filter("UserId", sap.ui.model.FilterOperator.Contains, sQuery));
```

Comment:
sap.ui.model.FilterOperator should be imported at the beginning of the file:

`
sap.ui.define([
		'./Formatter',
		'sap/ui/model/FilterOperator
	], function(Formatter, FilterOperator) {
...
aFilters.push(new Filter("UserId", FilterOperator.Contains, sQuery));
...
`